pkgs/development/python-modules/home-assistant-chip-wheels/default.nix

```diff
@@ -0,0 +1,305 @@
+  # with wheel files for all dependencies and point pip to it
+  # via its configuration file or environment variables.
+  # https://pigweed.dev/python_build.html#installing-offline
+  # The wheel of a python package foo is available as foo.dist.
+  # So that would be easy, but we also need wheels for transitive dependencies.
+  # the function saturateDependencies below computes this transitive closure.
+  #
+  # yes this list of dependencies contains both build tools and proper dependencies.
+  env.PIP_NO_INDEX = "1";
+  env.PIP_FIND_LINKS =
+    let
+      dependencies = [
+        appdirs
+        appnope
+        black
+        build
+        colorama
+        coloredlogs
+        coverage
+        click
+        cryptography
+        diskcache
+        googleapis-common-protos
+        google-cloud-storage
+        ipython
+        jinja2
+        json5
+        jsonschema
+        lark
+        mobly
+        mypy
+        mypy-extensions
+        mypy-protobuf
+        packaging
+        parameterized
+        pip-tools
+        prompt-toolkit
+        protobuf
+        psutil
+        ptpython
+        pyelftools
+        pygments
+        pykwalify
+        pylint
+        pyperclip
+        pyserial
+        python-daemon
+        pyyaml
+        requests
+        setuptools
+        six
+        sphinx
+        sphinx-argparse
+        sphinx-design
+        stringcase
+        toml
+        tornado
+        types-protobuf
+        types-pyyaml
+        types-requests
+        types-setuptools
+        watchdog
+        websockets
+        wheel
+        yapf
+      ];
+      depListToAttrs =
+        list:
+        builtins.listToAttrs (
+          map (dep: {
+            name = dep.name;
+            value = dep;
+          }) list
+        );
+      saturateDependencies =
+        deps:
+        let
+          before = deps;
+          new = lib.mergeAttrsList (
+            map (dep: depListToAttrs (dep.propagatedBuildInputs or [ ])) (lib.attrValues before)
+          );
+          after = before // new;
+        in
+        if lib.attrNames before != lib.attrNames after then saturateDependencies after else before;
+      saturateDependencyList = list: lib.attrValues (saturateDependencies (depListToAttrs list));
+      saturatedDependencyList = lib.filter (drv: drv ? dist) (saturateDependencyList dependencies);
+    in
+    lib.concatMapStringsSep " " (dep: "file://${dep.dist}") saturatedDependencyList;
+
+  configurePhase =
+    let
+      configureArguments = [
+        ''chip_project_config_include_dirs=["//.."]''
+        ''chip_crypto="boringssl"''
+        ''enable_rtti=true''
+        ''chip_config_memory_debug_checks=false''
+        ''chip_config_memory_debug_dmalloc=false''
+        ''chip_mdns="minimal"''
+        ''chip_minmdns_default_policy="libnl"''
+        ''chip_python_version="${lib.versions.majorMinor python3.version}"''
+        ''chip_python_platform_tag="any"''
```

Comment:
The [chip-wheels build script](https://github.com/home-assistant-libs/chip-wheels/blob/3a29da9a6b9734832177339d620c1ca10e37fb54/.github/workflows/build.yaml#L127) uses "manylinux_2_31". Do you know what the difference between "any" and "manylinux_2_31" is?

Reply:
First of all, the purpose of this tag is to download the correct precompiled wheel from a repository. Since nixpkgs automatically picks the right wheel without consulting this tag, it can be anything, I decided to use `any` because manylinux_2_31 is not correct on musl, or weird platforms.